api/src/main/scala/dx/api/DxApi.scala

```diff
@@ -892,17 +892,26 @@ case class DxApi(version: String = "1.0.0", dxEnv: DXEnvironment = DXEnvironment
     * Uploads a local file to the platform, and returns a DxFile.
     * Calls `dx upload` in a subprocess.
     */
-  def uploadFile(path: Path, destination: Option[String] = None, wait: Boolean = false): DxFile = {
+  def uploadFile(path: Path,
+                 destination: Option[String] = None,
+                 wait: Boolean = false,
+                 tags: Set[String] = Set.empty,
+                 properties: Map[String, String] = Map.empty): DxFile = {
     if (!Files.exists(path)) {
       throw new AppInternalException(s"Output file ${path.toString} is missing")
     }
 
     def uploadOneFile(path: Path): Option[String] = {
       try {
         // shell out to dx upload. We need to quote the path, because it may contain spaces
-        val destOpt = destination.map(d => s"""--destination "${d}" -p""").getOrElse("")
-        val waitOpt = if (wait) "--wait" else ""
-        val dxUploadCmd = s"""dx upload "${path.toString}" --brief ${destOpt} ${waitOpt}"""
+        val destOpt = destination.map(d => s""" --destination "${d}" -p""").getOrElse("")
+        val waitOpt = if (wait) " --wait" else ""
+        val tagsOpt = tags.map(tag => s" --tag ${tag}").mkString("")
+        val propertiesOpt = properties.map {
+          case (key, value) => s" --property ${key}=${value}".mkString("")
```

Comment:
This map() puts out a collection, so the final command string is going to have something like "List( --property a=A,  --property b=B)" in it

I've tried this in a sandbox project 
```
	val m = Map("a" -> "A", "b" -> "B")

	val s = m.map {
		case (key, value) => s" --property ${key}=${value}".mkString("")
	}

	val s1 = s"""start ${s} end""" 

	println(s1)
```

Reply:
@jdidion   Whereas you probably want to do
```
	val m = Map("a" -> "A", "b" -> "B")

	val s = m.map {
		case (key, value) => s" --property ${key}=${value}"
	}

	val s1 = s"""start ${s.mkString("")} end""" 

	println(s1)
```

Reply:
yep good catch